src/utils/utils.js

```diff
@@ -226,6 +226,21 @@ export function getActiveTab(callback) {
 	});
 }
 
+/**
+ * Query the first tab that matches the url argument
+ * @memberOf BackgroundUtils
+ *
+ * @param {function} callback		function to call if tab found
+ * @param {string} url				the tab url to search for
+ */
+export function getTabByUrl(callback, url) {
+	chrome.tabs.query({
+		url
+	}, (tabs) => {
+		callback(tabs[0]);
```

Comment:
@wlycdgrfromflatiron can you double check what the behavior here is if there are no tabs found with the query search criteria? If so, what would the value of tabs be? We might need to add a check so that we don't run into an "array out of bounds error" or "can't access variable of undefined" or something like that.

Reply:
If no tabs were found, tabs would be undefined, causing an error on the bracket access. I modified the code to guard against this in the commit below.